circuits/cpp/src/aztec3/circuits/kernel/public/.test.cpp

```diff
@@ -163,15 +162,15 @@ PublicKernelInputsNoPreviousKernel<NT> get_kernel_inputs_no_previous_kernel()
     const NT::fr portal_contract_address = 23456;
 
     const NT::address msg_sender = NT::fr(1);
-    const NT::address tx_origin = msg_sender;
+    const NT::address& tx_origin = msg_sender;
```

Comment:
Should this be a reference? If msg_sender changes, we don't want tx_origin to change...
Maybe it's ok in this particular case, because tx_origin is probably only used below to initialise structs... and I think (???) it will always create a copy of the value of tx_origin, rather than passing a reference.

Reply:
It shouldn't be a problem here because they are both const I believe. But yes you make a good point about it adding `&` in general. I'll look through and see if there are any sketchy spots where it did this.